src/QMCHamiltonians/ECPComponentBuilder.cpp

```diff
@@ -207,6 +207,43 @@ bool ECPComponentBuilder::put(xmlNodePtr cur)
   }
   if (pp_nonloc)
   {
+    if (Nrule == -1)
+    {
+      app_log() << "Quadrature rule not determined from pseudopotential xml file. Setting sensible default" << std::endl;
```

Comment:
Move after the switch and print which nrule was chosen.

Reply:
Please move the whole switch to a function.

Reply:
@ye-luo I think this is fine as is. Function won't help much, just more indirection for the reader. A strong argument for a function would be needing this switch more than once. However this is init only and we don't.

Reply:
I moved the print statement. I can change it to a function if necessary